### PR TITLE
Documentation updates for the PojoCodec updates

### PR DESCRIPTION
**Pull request by rozza**
JAVA-2686

### CODE REVIEW

File: docs/reference/content/bson/pojos.md [hunk 1]
```diff
@@ -196,6 +196,79 @@ all serializable because they are bound to the concrete types `Integer`, `String
 On their own, instances of `GenericTree` or `GenericClass` are not serializable by the `PojoCodec`. This is because the runtime type parameter
 information is erased by the JVM, and the type parameters cannot be specialized accurately.
 
+The 3.6 release of the driver further improves generic support with the addition of PropertyCodecProviders. The `PropertyCodecProvider` API 
+allows type safe support of container types, by providing concrete type parameters for the generic types as declared in the POJO.
```
Comment (jyemin) [inline docs/reference/content/bson/pojos.md]: Hyphen between "type" and "safe". 
Remove comma between "types" and "by"

File: docs/reference/content/bson/pojos.md [hunk 2]
```diff
@@ -196,6 +196,79 @@ all serializable because they are bound to the concrete types `Integer`, `String
...
+
+A great use of the `PropertyCodecProvider` API could be to add support for the Java 8 `Optional` class. The following example creates a 
+`OptionalPropertyCodecProvider`:
+
+```java
+public class OptionalPropertyCodecProvider implements PropertyCodecProvider {
```
Comment (jyemin) [inline docs/reference/content/bson/pojos.md]: Use Guava's [Optional](https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Optional.html) class as the example instead, as it's less likely that the driver would ever support that out of the box.

File: docs/reference/content/whats-new.md [hunk 3]
```diff
@@ -40,6 +40,18 @@ The 3.6 release adds support for [causally consistency](http://dochub.mongodb.or
...
 [`MongoClientOptions`]({{<apiref "com/mongodb/MongoClientOptions">}}).
 
+### PojoCodec improvements
+
+The 3.6 release brings new improvements to the `PojoCodec`:
+
+  * Improved sub-class and discriminator support.
+  * Support for custom Collection and Map implementations.
+  * Improvements to the `BsonCreator` annotation, which now supports `@BsonId` and `@BsonProperty` with values that represent the read name of the property.
+  * A new [`PojoCodecProvider`]({{<apiref "org/bson/codecs/pojo/PojoCodecProvider">}}) API, allowing for easy and type safe handling of container types.
```
Comment (jyemin) [inline docs/reference/content/whats-new.md]: Did you mean PropertyCodecProvider?

Comment (jyemin) [reply docs/reference/content/whats-new.md]: Also, hyphenate "type safe" here too.

File: docs/reference/content/bson/pojos.md (hunk 1 above)
Comment (rozza) [reply docs/reference/content/bson/pojos.md]: Done

File: docs/reference/content/bson/pojos.md (hunk 2 above)
Comment (rozza) [reply docs/reference/content/bson/pojos.md]: Done

File: docs/reference/content/whats-new.md (hunk 3 above)
Comment (rozza) [reply docs/reference/content/whats-new.md]: Yes! Updated

**Pull request closed without merging**